docarray/index/backends/elastic.py

```diff
@@ -0,0 +1,546 @@
+import os
+import uuid
+import warnings
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import (
+    Any,
+    Dict,
+    Generator,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import numpy as np
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import parallel_bulk
+from pydantic import parse_obj_as
+
+import docarray.typing
+from docarray import BaseDocument
+from docarray.index.abstract import (
+    BaseDocumentIndex,
+    _ColumnInfo,
+    _FindResultBatched,
+    _raise_not_composable,
+)
+from docarray.typing import AnyTensor
+from docarray.typing.tensor.ndarray import NdArray
+from docarray.utils.find import _FindResult
+from docarray.utils.misc import is_tf_available, is_torch_available
+
+TSchema = TypeVar('TSchema', bound=BaseDocument)
+T = TypeVar('T', bound='ElasticDocIndex')
+
+ELASTIC_PY_VEC_TYPES: List[Any] = [list, tuple, np.ndarray]
+
+if is_torch_available():
+    import torch
+
+    ELASTIC_PY_VEC_TYPES.append(torch.Tensor)
+
+if is_tf_available():
+    import tensorflow as tf  # type: ignore
+
+    from docarray.typing import TensorFlowTensor
+
+    ELASTIC_PY_VEC_TYPES.append(tf.Tensor)
+    ELASTIC_PY_VEC_TYPES.append(TensorFlowTensor)
+
+
```

Comment:
Do you think this index could inherit frome the v8 index and only change the methods and properties that need to be changed, so that we can share more code? Or is the difference between the two too big?

Reply:
v8 has more specific API (create, search, put_mapping etc.). For example
```python
body = {
    'size': limit,
    'query': {
        'bool': {
            'must': {'match': {search_field: query}},
        }
    },
}

#v7
resp = self._client.search(index=self._index_name,body=body,)
#v8
resp = self._client.search(index=self._index_name, query=search_query, size=limit,)
```

But I think the difference is not too big, they share many common helper functions.

Reply:
ok sounds good, we can optimize code re-use once this one is merged